labs/03_neural_recsys/Explicit_Feedback_Neural_Recommender_System.ipynb

```diff
@@ -460,28 +465,39 @@
    "source": [
     "# For each sample we input the integer identifiers\n",
     "# of a single user and a single item\n",
-    "user_id_input = Input(shape=[1], name='user')\n",
-    "item_id_input = Input(shape=[1], name='item')\n",
-    "\n",
-    "embedding_size = 30\n",
-    "user_embedding = Embedding(output_dim=embedding_size, input_dim=max_user_id + 1,\n",
-    "                           input_length=1, name='user_embedding')(user_id_input)\n",
-    "item_embedding = Embedding(output_dim=embedding_size, input_dim=max_item_id + 1,\n",
-    "                           input_length=1, name='item_embedding')(item_id_input)\n",
-    "\n",
-    "# reshape from shape: (batch_size, input_length, embedding_size)\n",
-    "# to shape: (batch_size, input_length * embedding_size) which is\n",
-    "# equal to shape: (batch_size, embedding_size)\n",
-    "user_vecs = Flatten()(user_embedding)\n",
-    "item_vecs = Flatten()(item_embedding)\n",
-    "\n",
-    "input_vecs = Concatenate()([user_vecs, item_vecs])\n",
-    "input_vecs = Dropout(0.99)(input_vecs)\n",
-    "\n",
-    "x = Dense(64, activation='relu')(input_vecs)\n",
-    "y = Dense(2, activation='tanh')(x)\n",
-    "\n",
-    "model = Model(inputs=[user_id_input, item_id_input], outputs=y)\n",
+    "class DeepRecoModel(Model):\n",
```

Comment:
I would rather call it `DeepRegressionModel` for consistency as even the implicit feedback models as recommenders.

Alternatively we could have:

- `SimpleExplicitFeedbackModel`.
- `DeepExplicitFeedbackModel`,
- `SimpleImplicitFeedbackModel`,
- `DeepImplicitFeedbackModel`.

as you prefer. I am also fine with `RegressionModel` / `DeepRegressionModel` for explicit feedback and then `TripletModel` / `DeepTripletModel` for the implicit feedback.